tests/fork/pool.spec.ts

```diff
@@ -0,0 +1,190 @@
+    expect(aTokensBalance).to.be.gte(depositSize);
+  });
+
+  it("Borrow Variable DAI and replay with DAI", async () => {
+    const { users, dai, aDai, pool, variableDebtDai } =
+    testEnv;
+    
+    const borrowSize = BigNumber.from(parseEther("1"));
+    const repaySize = borrowSize.add(borrowSize.mul(5).div(100));
+    const user = users[1];
+
+    const poolService = await pool.connect(user.signer);
+
+    await supply(dai, user, depositSize);
+
+    const aTokensBalance = await aDai.balanceOf(user.address);
+
+    expect(aTokensBalance).to.be.gt(zero);
+    expect(aTokensBalance).to.be.gte(depositSize);
+
+    // Set collateral
+    await waitForTx(
+      await poolService.setUserUseReserveAsCollateral(dai.address, true)
+    );
+
+    // Borrow DAI with DAI as collateral
+    await waitForTx(
+      await poolService.borrow(dai.address, borrowSize, RateMode.VARIABLE, "0", user.address)
+    );
+
+    const debtBalance = await variableDebtDai.balanceOf(user.address);
+
+    expect(debtBalance).to.be.gt(zero);
+
+    // Partial Repay DAI loan
+    const partialPayment = repaySize.div(2);
+
+    await repay(dai, user, partialPayment, RateMode.VARIABLE);
+    const debtBalanceAfterPartialRepay = await variableDebtDai.balanceOf(
+      user.address
+    );
+
+    expect(debtBalanceAfterPartialRepay).to.be.lt(debtBalance);
+  });
+
+  it.skip("Supply Stable DAI, borrow USDC and replay with DAI", async () => {
+    const { deployer, users, dai, usdc, aUsdc, pool, variableDebtDai, stableDebtDai } = testEnv;
+    
+    const mode: RateMode = RateMode.STABLE as RateMode;
+    const debtDai = mode === RateMode.VARIABLE ? variableDebtDai : stableDebtDai;
+    const borrowSize = BigNumber.from(parseEther("1"));
+    const repaySize = borrowSize.add(borrowSize.mul(5).div(100));
+    const user = users[1];
+
+    const poolService = await pool.connect(user.signer);
+
+    // Approve and Deposit with USDC
+    await supply(dai, user, depositSize);
+    await supply(usdc, deployer, depositSize);
+
+    const aTokensBalance = await aUsdc.balanceOf(user.address);
+
+    expect(aTokensBalance).to.be.gt(zero);
+    expect(aTokensBalance).to.be.gte(depositSize);
+
+    // Set collateral
+    await waitForTx(
+      await poolService.setUserUseReserveAsCollateral(dai.address, true)
+    );
+
+    // Borrow DAI with USDC as collateral
+    await waitForTx(
+      await poolService.borrow(usdc.address, borrowSize, mode, "0", user.address)
+    );
+
+    const debtBalance = await debtDai.balanceOf(user.address);
+    expect(debtBalance).to.be.gt(zero);
+
+    // Partial Repay DAI loan
+    const partialPayment = repaySize.div(2);
+
+    await repay(dai, user, partialPayment, mode);
+    
+    const debtBalanceAfterPartialRepay = await debtDai.balanceOf(
+      user.address
+    );
+    expect(debtBalanceAfterPartialRepay).to.be.lt(debtBalance);
+  });
+
+  it("Supply Variable DAI, borrow USDC and replay with DAI", async () => {
```

Comment:
You can in a beforeAll, or beforeEach as well set up some initial pool supply to avoid the need to supply in each test that is testing the repayment. Maybe even have an account with a debt already